snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java

```diff
@@ -99,7 +102,11 @@ public String getName() {
 
   @Override
   protected void onActorStarting() {
-    currentPersistedSnapshotRef.set(loadLatestSnapshot(snapshotsDirectory));
+    final FileBasedSnapshot latestSnapshot = loadLatestSnapshot(snapshotsDirectory);
+    currentPersistedSnapshotRef.set(latestSnapshot);
+    if (latestSnapshot != null) {
+      availableSnapshots.add(latestSnapshot);
```

Comment:
:wrench: :question: It's probably not necessary but for completeness sake: We could populate `availableSnapshots` in `loadLatestSnapshot()` instead, there we already load all snapshots. Then we'd have all snapshots in `availableSnapshots`, not just the latest one.

Reply:
On startup, we delete all older snapshots and keep only the latest one. So not sure if it is valuable to add all snapshots in available snapshots.

Reply:
Oh yeah, forgot about that :+1: